compiler/il/OMRNode.cpp

```diff
@@ -2316,17 +2316,115 @@ OMR::Node::dontEliminateStores(bool isForLocalDeadStore)
 bool
 OMR::Node::isNotCollected()
    {
-   TR::Compilation * comp = TR::comp();
-   if (self()->getOpCode().hasSymbolReference() && self()->getSymbolReference()->getSymbol()->isNotCollected())
-      return true;
-   else if (comp->generateArraylets() &&
-            (self()->getOpCode().isArrayRef() &&
-             (self()->getFirstChild()->isNotCollected() || !self()->getFirstChild()->isInternalPointer())))
-      return true;
-   return false;
+   return !self()->computeIsCollectedReference();
    }
 
 
+/**
+ * \brief
+ *    Check if the node is an internal pointer.
+ *
+ * \return
+ *    Return true if the node is an internal pointer, false otherwise.
+ *
+ * \note
+ *    This query currently only works on opcodes that can have a pinning array pointer,
+ *    i.e. aiadd, aiuadd, aladd, aluadd. Suppport for other opcodes can be added if necessary.
```

Comment:
I think aload and aRegload opcodes can also have a pinning array pointer. It seems to me the comment (and possibly the code) needs to be changed to ensure that those opcodes are included.